# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/nsIPrincipal.idl
# Commit: 89e396b2896e
# Full Hash: 89e396b2896ed7fc168a5b4ec2f8ead05f5c6802
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-06-03 04:20:58
# Regressor Bug: 1638358
# File Overlap Count: 1
# Description:
#   Bug 1638358 - Cookie Schemeful Same-Site - part 5 - schemeful comparison, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75629
# ==============================================================================

diff -r 33ad5fa05209 -r 89e396b2896e caps/nsIPrincipal.idl
--- a/caps/nsIPrincipal.idl	Tue Jun 02 09:14:09 2020 +0000
+++ b/caps/nsIPrincipal.idl	Tue Jun 02 09:15:24 2020 +0000
@@ -287,7 +287,12 @@
     /* Returns the Spec of the Principals URI with
     * user/pass/ref/query stripped for privacy and spoof prevention
     */
-     readonly attribute ACString exposableSpec;
+    readonly attribute ACString exposableSpec;
+
+    /**
+     * Return the scheme of the principals URI
+     */
+    readonly attribute ACString scheme;
 
     /**
     * Checks if the Principal's URI Scheme matches with the parameter